src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantAGoodErrorExperience.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.NET.TestFramework;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.Build.Tasks;
+using System.Reflection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenWeWantAGoodErrorExperience : SdkTest
+    {
+        public GivenWeWantAGoodErrorExperience(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [Fact]
+        public void ConfirmNoDuplicateErrorCodes()
+        {
+            Type type = typeof(Strings);
+            var resourceStrings = type.GetRuntimeProperties();
+            List<string> listOfErrorCode = new List<string>();
+            foreach (var resource in resourceStrings )
+            {
+                if ( resource.PropertyType == typeof(string))
+                {
+                    string errorCode = Strings.GetResourceString(resource.Name).Substring(0, 10);
```

Comment:
Doesn't this throw if you have a string of length < 10? Maybe just check errorCode.StartsWith("NETSDK") then from there assume it's at least 10 characters long and add characters 6-10?